backend/models/loginRecord.js

```diff
@@ -0,0 +1,127 @@
+                loginRecord.referrer = referrer;
+        }
+
+        const location = await getLocationFromIPAddress(loginRecord.ipAddr);   
+        loginRecord.location = {
+                country: location.country_name,
+                city: location.city
+        };    
+       
+         const userAgentString = req.headers['user-agent'];
+         var uaInfo = {};
+
+         if(isUserAgentFromPlugin(userAgentString)){
+                uaInfo = getUserAgentInfoFromPlugin(userAgentString);
+         }
+         else{
+                uaInfo = getUserAgentInfoFromBrowser(userAgentString);
+         }
+
+         loginRecord.application = uaInfo.application;
+         loginRecord.engine = uaInfo.engine;
+         loginRecord.os = uaInfo.os;
+         loginRecord.device = uaInfo.device;          
+
+         const col = await db.getCollection("loginRecords", req.body.username);
+         await col.insertOne(loginRecord);
+ }
+ 
+ //Format: 
+ //PLUGIN: {OS Name}/{OS Version} {Host Software Name}/{Host Software Version} {Plugin Type}/{Plugin Version}
+ //Example:
+ //PLUGIN: Windows/10.0.19042.0 REVIT/2021.1 PUBLISH/4.15.0
+ function getUserAgentInfoFromPlugin(userAgentString){
+        const userAgentInfo = {};        
+        const userAgentComponents = userAgentString.replace("PLUGIN: ", "").split(' ');                  
+
+        userAgentInfo.application = {
+                name: userAgentComponents[1].split('/')[0],
+                version: userAgentComponents[1].split('/')[1],
+                type: "plugin"        
+        }
+        
+        userAgentInfo.engine = {
+                name: "3drepoplugin",
+                version: userAgentComponents[2].split('/')[1]
+        }
+
+        userAgentInfo.os = {
+                osName: userAgentComponents[0].split('/')[0],
+                osVersion: userAgentComponents[0].split('/')[1]
+        }
+
+        userAgentInfo.device = "desktop";
+
+        return userAgentInfo;
+ }
+
+ function getUserAgentInfoFromBrowser(userAgentString){
+        const userAgentInfo = {};        
+        const userAgentObject = uaParserJs(userAgentString);       
+
+        userAgentInfo.application = {
+                name: userAgentObject.browser.name,
+                version: userAgentObject.browser.version,
+                type: "browser"        
+        }
+        
+        userAgentInfo.engine = {
+                name: userAgentObject.engine.name,
+                version: userAgentObject.engine.version
+        }
+
+        userAgentInfo.os = {
+                osName: userAgentObject.os.name,
+                osVersion: userAgentObject.os.version
+        }
+
+        userAgentInfo.device = device(userAgentString).type;
+
+        return userAgentInfo;
+}
+
+async function getLocationFromIPAddress(ipAddress){
+        return await ip2location.fetch(ipAddress);
+}
+
+function isUserAgentFromPlugin(userAgent){        
+        return userAgent.split(" ")[0] == "PLUGIN:";
```

Comment:
You almost never use `==` in javascript - there's a very small amount of situations where you find yourself needing to use `==` instead of `===` legitimately

```suggestion
        return userAgent.split(" ")[0] === "PLUGIN:";
```